File: bushop-scraper/app/schemas/route.py
from __future__ import annotations
from pydantic import BaseModel, ConfigDict


class ProvinceI18n(BaseModel):
    vi: str
    en: str
    ja: str


class RouteResponse(BaseModel):
    id: int
    slug: str
    origin_id: int | None = None
    destination_id: int | None = None
    origin_name: ProvinceI18n
    destination_name: ProvinceI18n

    model_config = ConfigDict(from_attributes=True)

File: bushop-src/src/types/route.ts
// src/types/route.ts
import type { ApiResponse } from './api';

export interface ProvinceI18n {
  vi: string;
  en: string;
  ja: string;
}

export interface Route {
  id: number;
  slug: string;
  origin_id: number;
  destination_id: number;
  origin_name: ProvinceI18n;
  destination_name: ProvinceI18n;
}

export type RoutesResponse = ApiResponse<Route[]>;
